In [1]:
import pandas as pd
import numpy as np

# Chargement du dataset
df = pd.read_csv('/content/survey.csv')

In [5]:
df['treatment_numeric'] = df['treatment'].map({'Yes': 1, 'No': 0})
bins = [18, 25, 35, 45, 55, 75]
labels = ['18-25', '26-35', '36-45', '46-55', '56-75']
df['age_group'] = pd.cut(df['Age'], bins=bins, labels=labels, right=False)

age_dist = df.groupby('age_group')['treatment_numeric'].mean() * 100
print("Pourcentage to seek treatment for mental health by age :")
print(age_dist)

Pourcentage to seek treatment for mental health by age :
age_group
18-25    44.871795
26-35    48.939180
36-45    54.687500
46-55    56.862745
56-75    70.588235
Name: treatment_numeric, dtype: float64


/tmp/ipykernel_1079/3734686523.py:6: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  age_dist = df.groupby('age_group')['treatment_numeric'].mean() * 100


In [7]:
gender_map = {

    'male': 'Male',
    'm': 'Male',
    'male-ish': 'Male',
    'maile': 'Male',
    'cis male': 'Male',
    'mal': 'Male',
    'male (cis)': 'Male',
    'make': 'Male',
    'man': 'Male',
    'msle': 'Male',
    'mail': 'Male',
    'malr': 'Male',
    'cis man': 'Male',

    'female': 'Female',
    'f': 'Female',
    'trans-female': 'Female',
    'cis female': 'Female',
    'woman': 'Female',
    'femake': 'Female',
    'cis-female/femme': 'Female',
    'female (cis)': 'Female',
    'femail': 'Female'
}

df['Gender_Cleaned'] = df['Gender'].str.lower().str.strip().map(gender_map)

df['Gender_Cleaned'] = df['Gender_Cleaned'].fillna('Other')

print(df['Gender_Cleaned'].value_counts())

Gender_Cleaned
Male      991
Female    248
Other      20
Name: count, dtype: int64


In [9]:
gender_dist = df.groupby('Gender_Cleaned')['treatment_numeric'].mean() * 100
print("\nPourcentage to seek treatment for mental health by gender :")
print(gender_dist)


Pourcentage to seek treatment for mental health by gender :
Gender_Cleaned
Female    68.548387
Male      45.408678
Other     85.000000
Name: treatment_numeric, dtype: float64


In [12]:
countries_count = df['Country'].value_counts()
major_countries = countries_count[countries_count >= 10].index
df_major_countries = df[df['Country'].isin(major_countries)]

country_dist = df_major_countries.groupby('Country')['treatment_numeric'].mean() * 100
country_dist = country_dist.sort_values(ascending=False)

print("\nHighest Rates :")
print(country_dist.head(3))

print("\nLowest Rates :")
print(country_dist.tail(3))


Highest Rates :
Country
Australia        61.904762
United States    54.593875
Canada           51.388889
Name: treatment_numeric, dtype: float64

Lowest Rates :
Country
Netherlands    33.333333
India          30.000000
France         15.384615
Name: treatment_numeric, dtype: float64


**Part 1 : Introduction to the Dataset**

1. Mental Health Distribution by Age GroupIn the tech industry, the tendency to seek treatment for mental health conditions shows a clear correlation with age. While younger professionals (18-25) seek treatment less frequently, nearly 71% of employees aged 56-75 have reported seeking professional help. This might suggest that long-term career pressure or a decrease in stigma among senior staff impacts these numbers.

2. Frequency of Mental Health Issues by GenderGender plays a significant role in the frequency of reported mental health issues. Female and non-binary ("Other") employees report seeking treatment at much higher rates than their male counterparts.
- Other/Non-binary: ~82.4% rate of treatment.
- Female: ~68.5% rate of treatment.
- Male: ~45.4% rate of treatment.

3. Geographical Trends (Highest vs Lowest Rates)By filtering for countries with a significant number of respondents (minimum 10), we identified regional differences in mental health reporting:Highest Rates: Australia (61.9%) and the United States (54.7%) lead the reported rates of treatment.Lowest Rates: France (15.4%) and India (30.0%) show the lowest rates, which may be influenced by different cultural stigmas or variations in corporate healthcare benefits.

In [ ]:
#Part 2: Data Cleaning

import pandas as pd

df = pd.read_csv('/content/survey.csv')

#Dealing with missing values
missing_counts = df.isnull().sum()
print(df.shape)
missing_pct = (missing_counts / len(df)) * 100
missing_df = pd.DataFrame(
    {
        'Missing Count': missing_counts,
        'Missing %': missing_pct
    })
print(missing_df)

df = df.drop(columns=['comments'])
df.dropna(subset=['self_employed'], inplace=True)
df['work_interfere'] = df['work_interfere'].fillna('Unknown')
df['state'] = df['state'].fillna('Unknown')
print(df.isnull().sum())

# # we keep only the lines with ages from 18 to 75
# df = df[(df['Age'] >= 18) & (df['Age'] <= 75)]



(1259, 27)
                           Missing Count  Missing %
Timestamp                              0   0.000000
Age                                    0   0.000000
Gender                                 0   0.000000
Country                                0   0.000000
state                                515  40.905481
self_employed                         18   1.429706
family_history                         0   0.000000
treatment                              0   0.000000
work_interfere                       264  20.969023
no_employees                           0   0.000000
remote_work                            0   0.000000
tech_company                           0   0.000000
benefits                               0   0.000000
care_options                           0   0.000000
wellness_program                       0   0.000000
seek_help                              0   0.000000
anonymity                              0   0.000000
leave                                  0   0.000000
m

Gender_Cleaned
Male      978
Female    243
Other      20
Name: count, dtype: int64


In [ ]:
df.drop(columns=['Timestamp'], inplace=True)
print(df.isnull().sum())
# Final check
print(f"Cleaned dataset contains {df.shape[0]} rows and {df.shape[1]} columns.")
print(df.head())

Age                          0
Gender                       0
Country                      0
state                        0
self_employed                0
family_history               0
treatment                    0
work_interfere               0
no_employees                 0
remote_work                  0
tech_company                 0
benefits                     0
care_options                 0
wellness_program             0
seek_help                    0
anonymity                    0
leave                        0
mental_health_consequence    0
phys_health_consequence      0
coworkers                    0
supervisor                   0
mental_health_interview      0
phys_health_interview        0
mental_vs_physical           0
obs_consequence              0
Gender_Cleaned               0
dtype: int64
Cleaned dataset contains 1241 rows and 26 columns.
    Age Gender        Country    state self_employed family_history treatment  \
18   46   male  United States       MD           